spec/index.bs

```diff
@@ -1276,30 +1288,32 @@ See {{GPUAdapter/limits|GPUAdapter.limits}} and {{GPUDevice/limits|GPUDevice.lim
-    readonly attribute unsigned long maxDynamicUniformBuffersPerPipelineLayout;
-    readonly attribute unsigned long maxDynamicStorageBuffersPerPipelineLayout;
-    readonly attribute unsigned long maxSampledTexturesPerShaderStage;
-    readonly attribute unsigned long maxSamplersPerShaderStage;
-    readonly attribute unsigned long maxStorageBuffersPerShaderStage;
-    readonly attribute unsigned long maxStorageTexturesPerShaderStage;
-    readonly attribute unsigned long maxUniformBuffersPerShaderStage;
-    readonly attribute unsigned long long maxUniformBufferBindingSize;
-    readonly attribute unsigned long long maxStorageBufferBindingSize;
-    readonly attribute unsigned long minUniformBufferOffsetAlignment;
-    readonly attribute unsigned long minStorageBufferOffsetAlignment;
-    readonly attribute unsigned long maxVertexBuffers;
-    readonly attribute unsigned long maxVertexAttributes;
-    readonly attribute unsigned long maxVertexBufferArrayStride;
-    readonly attribute unsigned long maxInterStageShaderComponents;
-    readonly attribute unsigned long maxComputeWorkgroupStorageSize;
-    readonly attribute unsigned long maxComputeInvocationsPerWorkgroup;
-    readonly attribute GPUExtent3D maxComputeWorkgroupDimensions;
-    readonly attribute unsigned long maxComputeWorkgroupsPerDimension;
+    readonly attribute GPUSize32 maxTextureDimension1D;
+    readonly attribute GPUSize32 maxTextureDimension2D;
+    readonly attribute GPUSize32 maxTextureDimension3D;
+    readonly attribute GPUSize32 maxTextureArrayLayers;
+    readonly attribute GPUSize32 maxBindGroups;
+    readonly attribute GPUSize32 maxDynamicUniformBuffersPerPipelineLayout;
+    readonly attribute GPUSize32 maxDynamicStorageBuffersPerPipelineLayout;
+    readonly attribute GPUSize32 maxSampledTexturesPerShaderStage;
+    readonly attribute GPUSize32 maxSamplersPerShaderStage;
+    readonly attribute GPUSize32 maxStorageBuffersPerShaderStage;
+    readonly attribute GPUSize32 maxStorageTexturesPerShaderStage;
+    readonly attribute GPUSize32 maxUniformBuffersPerShaderStage;
+    readonly attribute GPUSize64 maxUniformBufferBindingSize;
+    readonly attribute GPUSize64 maxStorageBufferBindingSize;
+    readonly attribute GPUSize32 minUniformBufferOffsetAlignment;
+    readonly attribute GPUSize32 minStorageBufferOffsetAlignment;
+    readonly attribute GPUSize32 maxVertexBuffers;
+    readonly attribute GPUSize32 maxVertexAttributes;
+    readonly attribute GPUSize32 maxVertexBufferArrayStride;
+    readonly attribute GPUSize32 maxInterStageShaderComponents;
+    readonly attribute GPUSize32 maxComputeWorkgroupStorageSize;
+    readonly attribute GPUSize32 maxComputeInvocationsPerWorkgroup;
+    readonly attribute GPUSize32 maxComputeWorkgroupX;
+    readonly attribute GPUSize32 maxComputeWorkgroupY;
+    readonly attribute GPUSize32 maxComputeWorkgroupZ;
+    readonly attribute GPUSize32 maxComputeWorkgroupsPerDimension;
```

Comment:
I believe `[EnforceRange]` is not valid on `readonly attribute`s which is also why we use `GPUFlagsConstant` instead of e.g. `GPUBufferUsageFlags` in `GPUBufferUsage`.

Reply:
Ah! I seem to recall bumping into this previously as well. I'll revert this part.